src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,72 @@
+package parser;
+
+import commands.*;
+import exceptions.InvalidCommandException;
+import exceptions.InvalidDescriptionException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    /**
+     * Interprets the user's input and returns the respective command.
+     * @param command user's input string
+     * @return command according to user's input
+     * @throws InvalidDescriptionException empty description of task
+     * @throws InvalidCommandException unknown command
+     */
+    public static Command parse(String command) throws InvalidDescriptionException, InvalidCommandException {
+        String[] splitString = command.split(" ");
+        String taskType = splitString[0];
+        if (command.equals("list")) {
+            return new ListCommand();
+        } else if (splitString[0].equals("done")) {
+            int taskIndex = Integer.valueOf(splitString[1]) - 1;
+            return new DoneCommand(taskIndex);
+        } else if (command.equals("bye")) {
+            return new ByeCommand();
+        } else if (taskType.equals("todo")) {
+            String[] todoSplit = command.split("todo ");
+            if (todoSplit.length == 1 || todoSplit[1].isEmpty()) {
+                throw new InvalidDescriptionException("\u2639 OOPS!!! The description of a todo cannot be empty.");
```

Comment:
Why not check for correct desc of todo when initializing todo object ie. in todo constructor? Since its a exception arising from trying to create a todo object.

Reply:
Same for the other object creation related exceptions.